src/main/java/fi/dy/masa/servux/dataproviders/StructureDataProvider.java

```diff
@@ -1,94 +1,159 @@
+    private BlockPos spawnPos = BlockPos.ORIGIN;
+    private int spawnChunkRadius = -1;
+    private boolean refreshSpawnMetadata;
+
     protected StructureDataProvider()
     {
         super("structure_bounding_boxes",
-              StructureDataPacketHandler.CHANNEL, StructureDataPacketHandler.PROTOCOL_VERSION,
-              "Structure Bounding Boxes data for structures such as Witch Huts, Ocean Monuments, Nether Fortresses etc.");
-
-        this.metadata.putString("id", StructureDataPacketHandler.CHANNEL.toString());
+                ServuxStructuresHandler.CHANNEL_ID,
+                ServuxStructuresHandler.PROTOCOL_VERSION,
+    "Structure Bounding Boxes data for structures such as Witch Huts, Ocean Monuments, Nether Fortresses etc.");
+
+        this.metadata.putString("name", this.getName());
+        this.metadata.putString("id", this.getNetworkChannel().toString());
+        this.metadata.putInt("version", this.getProtocolVersion());
+        this.metadata.putString("servux", Reference.MOD_STRING);
         this.metadata.putInt("timeout", this.timeout);
-        this.metadata.putInt("version", StructureDataPacketHandler.PROTOCOL_VERSION);
+
+        // TODO --> Move out of structures channel in the future
+        this.metadata.putInt("spawnPosX", this.getSpawnPos().getX());
+        this.metadata.putInt("spawnPosY", this.getSpawnPos().getY());
+        this.metadata.putInt("spawnPosZ", this.getSpawnPos().getZ());
+        this.metadata.putInt("spawnChunkRadius", this.getSpawnChunkRadius());
     }
 
     @Override
-    public boolean shouldTick()
+    public void setEnabled(boolean toggle)
+    {
+        this.enabled = toggle;
+    }
+
+    @Override
+    public void registerHandler()
+    {
+        ServerPlayHandler.getInstance().registerServerPlayHandler(HANDLER);
+        HANDLER.registerPlayPayload(ServuxStructuresPayload.TYPE, ServuxStructuresPayload.CODEC, IPluginServerPlayHandler.BOTH_SERVER);
+        HANDLER.registerPlayReceiver(ServuxStructuresPayload.TYPE, HANDLER::receivePlayPayload);
+    }
+
+    @Override
+    public void unregisterHandler()
+    {
+        HANDLER.unregisterPlayReceiver();
+        ServerPlayHandler.getInstance().unregisterServerPlayHandler(HANDLER);
+    }
+
+    @Override
+    public IPluginServerPlayHandler<ServuxStructuresPayload> getPacketHandler()
     {
-        return true;
+        return HANDLER;
     }
 
     @Override
-    public IPluginChannelHandler getPacketHandler()
+    public boolean shouldTick()
     {
-        return StructureDataPacketHandler.INSTANCE;
+        return this.enabled;
     }
 
     @Override
     public void tick(MinecraftServer server, int tickCounter)
     {
         if ((tickCounter % this.updateInterval) == 0)
         {
-            if (this.registeredPlayers.isEmpty() == false)
+            //Servux.printDebug("=======================\n");
+            //Servux.printDebug("tick: %d - %s\n", tickCounter, this.isEnabled());
```

Comment:
should probably remove the commented code or log it at Slf4j/Log4j debug levels _(must be available in Minecraft)_

Reply:
Those are old debug messages from masa, all I changed is moving it from "System.out.println"